### ISSUE TITLE
 Inaccurate Transcription and Language Model Issues

### ISSUE THREAD

**Issue by romanokeser** [sandrohanea/whisper.net#143]
I have encountered significant challenges using whisper for speech-to-text conversion in the Croatian language. Unlike English, the system consistently produces inaccurate transcriptions when using Croatian audio inputs. There is no difference in outputs between `CreateBuilder().WithLanguage("auto")` and `CreateBuilder().WithLanguage("Croatian")` 

Any suggestions? Thanks!

**Comment by sandrohanea** [comment]
Hello @romanokeser ,

Indeed, @gorokhovskiy is right, this is just a wrapper of whisper.cpp which is a C++ port of OpenAI Whisper, and the models are coming from the open ai.

Besides the issue with the Serbian-Croation languages, I can offer some additional ideas on how to improbe the quality of transcripts (for any language):
 1. Use a larger model. Ofc, this will require more memory and the transcript will be slower but the quality will be improved.
 2. Finetune your own model. You can also finetune your own model for a specific language, but that's a little harder:
 
     1. Find a Croatian dataset with labels.
     2. Follow: https://huggingface.co/blog/fine-tune-whisper
     3. With the resulted model, follow: https://github.com/ggerganov/whisper.cpp/blob/master/models/README.md#fine-tuned-models to map it to ggml format
     4. Use that ggml model with Whisper.net
     5. Optional but nice => share the finetuned model with the community.
     
      
To answer your questions about `WithLanguage("auto")` vs `WithLanguage("Croatian")`:
auto will first run the language identification and will detect the language of your audio, transcribing (or translation) will be identical after the language identification phase. 

Shorter said: the `auto` will just make it a little slower until you will get the first results, but the quality will be the same.

Notes: it can be worse in case `auto` is detecting a different language (by error) => e.g. other Slavic language.
